packages/core/src/storage/IndyStorageService.ts

```diff
@@ -187,6 +187,22 @@ export class IndyStorageService<T extends BaseRecord> implements StorageService<
     }
   }
 
+  /** @inheritDoc */
+  public async deleteById(recordClass: BaseRecordConstructor<T>, id: string): Promise<void> {
+    try {
```

Comment:
Maybe we can modify delete() method as well, to call this one. Something like:

```ts
  public async delete(record: T) {
    return await this.deleteById(record.type, record.id)
  }
```

Reply:
@tipusingh have you seen this one? I think we can optimize a bit here, but otherwise I' OK

Reply:
I changed the deleteById method 
```diff
-    deleteById(recordClass: BaseRecordConstructor<T>, id: string): Promise<void>
+    deleteById(recordType: string, id: string): Promise<void>
``` 
suggest if we could use another approach

Reply:
The error does not need to be handled separately there. We just can just use the error-handling from the `deleteById()`.

Reply:
Resolved